tests/unit/Pipeline.cpp

```diff
@@ -657,7 +663,8 @@ BOOST_AUTO_TEST_CASE(SingleElementPipelineFailure) {
 
   auto Error = Pipeline.run("end", Targets);
   BOOST_TEST(!!Error);
-  consumeError(std::move(Error));
+  if (!!Error)
```

Comment:
The `!!` pattern seems suspicious to me. Can we avoid it or is there a reason?

Reply:
Ok, I was wrong, it is essential in `BOOST_TEST`s, as those do not convert the input type to bool AND they pass it as is into `operator<<` of the std-stream, which llvm types don't play nicely with.

I'd suggest a more clean approach to the error being checked and, more importantly, printed properly, but let's be honest, this is old pipeline tests - those go away soon anyway. I say let them be.